pkg/controller/prpqr_reconciler/prpqr_reconciler.go

```diff
@@ -0,0 +1,190 @@
+	if err != nil {
+		return fmt.Errorf("failed to construct controller: %w", err)
+	}
+
+	// Watch only on Create events
+	predicateFuncs := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Object.GetNamespace() == ns
+		},
+		DeleteFunc:  func(event.DeleteEvent) bool { return false },
+		UpdateFunc:  func(event.UpdateEvent) bool { return false },
+		GenericFunc: func(event.GenericEvent) bool { return false },
+	}
+	if err := c.Watch(source.NewKindWithCache(&prpqv1.PullRequestPayloadQualificationRun{}, manager.GetCache()), prpqrHandler(), predicateFuncs); err != nil {
+		return fmt.Errorf("failed to create watch for PullRequestPayloadQualificationRun: %w", err)
+	}
+
+	r.logger.Info("Successfully added reconciler to manager")
+	return nil
+}
+
+func prpqrHandler() handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(func(o ctrlruntimeclient.Object) []reconcile.Request {
+		prpqr, ok := o.(*prpqv1.PullRequestPayloadQualificationRun)
+		if !ok {
+			logrus.WithField("type", fmt.Sprintf("%T", o)).Error("Got object that was not a PullRequestPayloadQualificationRun")
+			return nil
+		}
+
+		return []reconcile.Request{
+			{NamespacedName: types.NamespacedName{Namespace: prpqr.Namespace, Name: prpqr.Name}},
+		}
+	})
+}
+
+type reconciler struct {
+	logger *logrus.Entry
+	client ctrlruntimeclient.Client
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	logger := r.logger.WithField("request", req.String())
+	err := r.reconcile(ctx, req, logger)
+	if err != nil {
+		logger.WithError(err).Error("Reconciliation failed")
+	} else {
+		logger.Info("Finished reconciliation")
+	}
+	return reconcile.Result{}, controllerutil.SwallowIfTerminal(err)
+}
+
+func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, logger *logrus.Entry) error {
+	logger = logger.WithField("namespace", req.Namespace).WithField("prpqr_name", req.Name)
+	logger.Info("Starting reconciliation")
+
+	prpqr := &prpqv1.PullRequestPayloadQualificationRun{}
+	if err := r.client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: req.Namespace, Name: req.Name}, prpqr); err != nil {
+		return fmt.Errorf("failed to get the PullRequestPayloadQualificationRun: %s in namespace %s: %w", req.Name, req.Namespace, err)
+	}
+
+	prowjobs := generateProwjobs(prpqr.Spec.PullRequest.Org, prpqr.Spec.PullRequest.Repo, prpqr.Spec.PullRequest.BaseRef, req.Name, req.Namespace)
+	for _, pj := range prowjobs {
+		logger = logger.WithFields(logrus.Fields{"name": pj.Name, "namespace": req.Namespace})
+
+		err := r.client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: pj.Namespace, Name: pj.Name}, &prowv1.ProwJob{})
+		if err != nil && !kerrors.IsNotFound(err) {
+			return fmt.Errorf("failed to get the PullRequestPayloadQualificationRun: %s in namespace %s: %w", req.Name, req.Namespace, err)
+		}
+
+		if kerrors.IsNotFound(err) {
```

Comment:
Maybe invert this to a guard clause (`if !IsNotFound { continue }`) and shift the code left?